app/services/db/deleted_user/create.rb

```diff
@@ -3,15 +3,22 @@ module DeletedUser
     class Create
       def self.call(user_id)
         user = User.find_by(id: user_id)
+        deleted_user = nil
         return unless user
-        ::DeletedUser.create!(
-          user_id: user.id,
-          uuid: user.uuid,
-          user_created_at: user.created_at,
-          deleted_at: Time.zone.now,
-        )
-      rescue ActiveRecord::RecordNotUnique
-        ::DeletedUser.where(user_id: user_id)&.first
+
+        begin
```

Comment:
I'm not seeing what this `begin` gives us? Since we're doing the `rescue...raise` inside the `transaction do...end` block the separate `begin` is not needed right?

Reply:
Thanks for catching that, dropped the `begin`